internal/pkg/watch/docker_logs.go

```diff
@@ -324,3 +330,32 @@ func (w *DockerLogWatch) StartUnsafe() {
 func (w *DockerLogWatch) Stop() {
 	w.Watch.Stop()
 }
+
+// PendingStart waits until node type is determined and calls
+// chain.LogWatchEnabled() to check if it should start up or not.
+func (w *DockerLogWatch) PendingStart(subscriptions ...chan<- interface{}) {
+	ticker := time.NewTicker(time.Second)
+
+	for {
+		select {
+		case <-w.StopKey:
+			return
+		case <-ticker.C:
+			nodeType := global.BlockchainNode.NodeRole()
+			if nodeType == "" {
+				continue
+			}
+			log := w.Log.With("node_type", nodeType)
+			if global.BlockchainNode.LogWatchEnabled() {
+				if err := DefaultWatchRegistry.RegisterAndStart(w, subscriptions...); err != nil {
+					log.Errorw("failed to register docker log watcher", zap.Error(err))
+					return
+				}
+				log.Info("docker log watch started")
```

Comment:
Missing `return`

Reply:
Could you please write a regression test for that? I only noticed that when running the agent on one of the consensus nodes on staging.

Reply:
Addressed in [cd8a971](https://github.com/Metrika-Inc/agent/pull/95/commits/cd8a9712ddfadd652e8a03b8ac88333007689ce2)